### ISSUE TITLE
How can I avoid duplicate messages when monitoring PACKET_IN on the POX?

### ISSUE THREAD

**Issue by MXuanyuan** [noxrepo/pox#224]
Hi, I've been stuck in this strange thing recently. When monitoring the PACKET_IN message on the controller, I notice there are duplicate packets occurring when only use tree topology. It looks like a single message occurs three times, this makes my datasets of experiments inaccurate, how can I solve such issue?

This is an example of the output:
           dst              src
0         10.0.0.18   141.242.206.16
1         10.0.0.18   141.242.206.16
2         10.0.0.18   141.242.206.16
3         10.0.0.27   234.216.221.73
4         10.0.0.27   234.216.221.73
5         10.0.0.27   234.216.221.73
6         10.0.0.34  120.234.204.232
7         10.0.0.34  120.234.204.232
8         10.0.0.34  120.234.204.232
9          10.0.0.3     77.106.212.7
10         10.0.0.7   14.203.161.184
11         10.0.0.7   14.203.161.184
12         10.0.0.7   14.203.161.184
13         10.0.0.6  250.129.106.138
14        10.0.0.16   66.134.234.208
15        10.0.0.16   66.134.234.208
16        10.0.0.16   66.134.234.208
...

**Comment by MXuanyuan** [comment]
``

**Comment by MurphyMc** [comment]
Is it because the packet traverses three switches and you're logging it on each one?

**Comment by MXuanyuan** [comment]
> Is it because the packet traverses three switches and you're logging it on each one?

Hi, thanks for replying. Actually, there were seven switches in my topology. I also tried to use less switches, but same issue occurred except for single sw in use. I'll try to provide more details later.

**Comment by MXuanyuan** [comment]
> Is it because the packet traverses three switches and you're logging it on each one?

Hey, I'm back. Thanks for replying. 
I just tested it again, this time it's perfectly normal but nothing has been changed, same topology, some scripts....


37         10.0.0.8  184.189.206.239
38         10.0.0.8   253.145.70.187
39         10.0.0.8   230.110.171.12
40         10.0.0.8  101.248.229.226
41         10.0.0.8  132.105.177.200
42         10.0.0.8   199.48.159.220

That's weird.

Anyway I'll do more testing and try to analyse whether the reason of switches number. 

Thanks again.
